src/tests/test_frodokem.cpp

```diff
@@ -0,0 +1,224 @@
+/*
+ * Tests for FrodoKEM ("You SHALL Pass")
+ * - KAT tests using the KAT vectors from
+ *   https://github.com/microsoft/PQCrypto-LWEKE/tree/master/KAT
+ *
+ * (C) 2023 Jack Lloyd
+ * (C) 2023 René Meusel and Amos Treiber, Rohde & Schwarz Cybersecurity
+ *
+ * Botan is released under the Simplified BSD License (see license.txt)
+ */
+
+#include "test_rng.h"
+#include "tests.h"
+
+#include <iostream>
+#include <iterator>
+#include <memory>
+
+#if defined(BOTAN_HAS_FRODOKEM)
+   #include "test_pubkey.h"
+
+   #include <botan/frodokem.h>
+   #include <botan/pubkey.h>
+   #include <botan/xof.h>
+   #include <botan/internal/fmt.h>
+   #include <botan/internal/frodo_constants.h>
+#endif
+
+namespace Botan_Tests {
+
+#if defined(BOTAN_HAS_FRODOKEM)
+
+namespace {
+
+   #if defined(BOTAN_HAS_AES)
+
+Botan::FrodoKEMMode get_mode(std::string_view header) {
+   if(header == "FrodoKEM-640-SHAKE") {
+      return Botan::FrodoKEMMode::FrodoKEM640_SHAKE;
+   } else if(header == "FrodoKEM-976-SHAKE") {
+      return Botan::FrodoKEMMode::FrodoKEM976_SHAKE;
+   } else if(header == "FrodoKEM-1344-SHAKE") {
+      return Botan::FrodoKEMMode::FrodoKEM1344_SHAKE;
+   } else if(header == "eFrodoKEM-640-SHAKE") {
+      return Botan::FrodoKEMMode::eFrodoKEM640_SHAKE;
+   } else if(header == "eFrodoKEM-976-SHAKE") {
+      return Botan::FrodoKEMMode::eFrodoKEM976_SHAKE;
+   } else if(header == "eFrodoKEM-1344-SHAKE") {
+      return Botan::FrodoKEMMode::eFrodoKEM1344_SHAKE;
+   } else if(header == "FrodoKEM-640-AES") {
+      return Botan::FrodoKEMMode::FrodoKEM640_AES;
+   } else if(header == "FrodoKEM-976-AES") {
+      return Botan::FrodoKEMMode::FrodoKEM976_AES;
+   } else if(header == "FrodoKEM-1344-AES") {
+      return Botan::FrodoKEMMode::FrodoKEM1344_AES;
+   } else if(header == "eFrodoKEM-640-AES") {
+      return Botan::FrodoKEMMode::eFrodoKEM640_AES;
+   } else if(header == "eFrodoKEM-976-AES") {
+      return Botan::FrodoKEMMode::eFrodoKEM976_AES;
+   } else if(header == "eFrodoKEM-1344-AES") {
+      return Botan::FrodoKEMMode::eFrodoKEM1344_AES;
+   }
+
+   BOTAN_ASSERT_UNREACHABLE();
```

Comment:
This is reachable at runtime if someone flubs the test data. It should probably throw `Test_Error`